lib/components/global_state.dart

```diff
@@ -0,0 +1,7 @@
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+
+class GlobalState {
+  final mediaPlaybackIsLoading = StateProvider<bool>((ref) => false);
```

Comment:
Using a provider has some side effects. E.g. if you press the play button and one of the request fails, the loading animation persists (which is not the problem in itself) But since the isLoading state is used globally it also affects the DetailScreen etc. I think it would be better to use a ValueNotifier that sets isLoading to true after the button has been pressed and sets it back to false just before going to the PlayerScreen

It's a bit more complicated for the image banner. I think we could add a new isLoading parameter to the Mobile and Desktop Version and the we could control the state from the outside.

That way we don't need to introduce a global state and don't have to make changes to the PlayerScreen.

Reply:
i will think about this. the reason a jellyfin client typically wants a global playback state is for the eventual implementation of global playlist management.  so that the state of playback and the queue can be checked from anywhere.

Reply:
my instinct on solving the problem you mention would be to just add a timer to the state.  if we reach the network timeout and nothing shut off the timer, log a playback error and shut it off anyway.